### ISSUE TITLE
Intermediate Container Incorrectly Assumes /bin/echo Available

### ISSUE THREAD

**Issue by alunduil** [docker/compose#919]
When using the etcd image from the CoreOS guys: quay.io/coreos/etcd, I get the following error from docker-compose:

```
Cannot start container a6da0c7e877b1075696d64802f7e159b7660e5cd395064b891ab5712c74bc266: exec: "/bin/echo": stat /bin/echo: no such file or directory
```

This is when starting the following datastore service from docker-compose.yml:

```
datastore:
  image: quay.io/coreos/etcd:latest
  ports:
   - "4001:4001"
```

I found https://github.com/docker/fig/blob/fd30920aac060b03d6e67ec5c3170f670a73d00e/compose/service.py#L288, which indicates that this assumption is hard coded and containers *must* provide /bin/echo in order to be usable by docker-compose.  Is there any way we can remove this assumption so a greater number of containers are usable?

**Comment by tianon** [comment]
This is related to the discussion in https://github.com/docker/fig/pull/517 :+1:

**Comment by bfirsh** [comment]
:+1: solution is https://github.com/docker/fig/issues/874 I believe

**Comment by tristan0x** [comment]
I encountered the same issue while using minimalist [tianon/true](https://registry.hub.docker.com/u/tianon/true/) Docker image that only provides `/true`utility.
I am using Docker 1.5 and Docker Compose 1.1.0.
You can use the following YAML file to reproduce the issue

```yaml
data:
  image: tianon/true
  volumes:
    - /data
```

```sh
$ docker-compose up
Creating tmp_data_1...
Attaching to tmp_data_1
tmp_data_1 exited with code 0
Gracefully stopping... (press Ctrl+C again to force)

$ docker-compose up
Recreating tmp_data_1...
Cannot start container     3c29600d4043393936497d39c9c8c355f2427f4a34c86ea65d4ed3d315c50960: exec: "/bin/echo": stat /bin/echo: no such file or directory
```

As I want to keep data-only container as small as possible, my workaround is yet another [cogniteev/truecho](https://registry.hub.docker.com/u/cogniteev/truecho/) Docker image based on `tianon/true` that provides `/bin/echo` executable.

**Comment by tj** [comment]
having this issue with nsqio/nsq as well

```
Client version: 1.5.0
Client API version: 1.17
Go version (client): go1.4.1
Git commit (client): a8a31ef
OS/Arch (client): darwin/amd64
Server version: 1.5.0
Server API version: 1.17
Go version (server): go1.4.1
Git commit (server): a8a31ef
```

```
docker-compose 1.1.0
```

**Comment by saks** [comment]
+1, the same problem here

**Comment by balboah** [comment]
+1

**Comment by balboah** [comment]
Always happens at 2nd `fig up` that uses nsqio/nsq container

**Comment by marcosnils** [comment]
Same issue here. Any update about when this will resolved?

**Comment by ZeissS** [comment]
+1

**Comment by laurent35240** [comment]
+1

**Comment by tbal** [comment]
+1

**Comment by MitchK** [comment]
+1

**Comment by marchev** [comment]
+1

I am also able to reproduce when trying to bring up containers based on the following images:
 - `tomcat:8-jre8` 
 - `mongo:3`

**Comment by aanand** [comment]
This will be fixed by https://github.com/docker/compose/pull/1349.

**Comment by morenoh149** [comment]
has this been fixed? I'm trying to run kubernetes with docker-compose.
```
# docker-compose.yml
etcd:  
  image: gcr.io/google_containers/etcd:2.0.9
  net: host
  command: ['/usr/local/bin/etcd', '--bind-addr=0.0.0.0:4001', '--data-dir=/var/etcd/data']
...
```
```
$ docker-compose up                           
Recreating gettingstarted_apiserver_1...
Recreating gettingstarted_kubelet_1...
Recreating gettingstarted_controller_1...
Recreating gettingstarted_proxy_1...
Recreating gettingstarted_etcd_1...
Cannot start container eceaf23e94a5fec0e0edea5623aa824e01c944d96b8815c0d4aacfa23d77d81f: [8] 
System error: exec: "/bin/echo": stat /bin/echo: no such file or directory
```

**Comment by dnephin** [comment]
Yup, this was fixed by #1349 which is in 1.3.x
